### PR TITLE
[CI] Add jazzy :rocket: 

### PR DESCRIPTION
**Pull request by christophfroehlich**
- Jazzy jobs + readme
- because all of the repos file are empty, there was no difference between binary and semi-binary. So I deleted also the semi-binary workflow file.

Jobs for jazzy-main are failing, but jazzy-testing works fine already.

### CODE REVIEW

File: realtime_tools-not-released.jazzy.repos [hunk 1]
```diff
@@ -0,0 +1,6 @@
+repositories:
+  ## EXAMPLE DEPENDENCY
+#   <some_ros_package>:
+#     type: git
+#     url: git@github.com:<some_github_namespace>/<some_ros_package>.git
+#     version: master
```
Comment (saikishor) [inline realtime_tools-not-released.jazzy.repos:6]: @christophfroehlich do we need these files?

File: realtime_tools.jazzy.repos [hunk 2]
```diff
@@ -0,0 +1,6 @@
+repositories:
+  ## EXAMPLE DEPENDENCY
+#   <some_ros_package>:
+#     type: git
+#     url: git@github.com:<some_github_namespace>/<some_ros_package>.git
+#     version: master
```
Comment (saikishor) [inline realtime_tools.jazzy.repos:6]: This one too?

**Review by saikishor — COMMENTED** [review]
Rest looks good to me :)

**Comment by christophfroehlich** [comment]
@saikishor I haven't questioned this, just added jazzy. But you are right, it works without specifying a repos file. And because all of them are empty, there was no difference between binary and semi-binary. So I deleted also the semi-binary workflow file.

File: .github/workflows/ci-coverage-build.yml [hunk 3]
```diff
@@ -11,7 +11,7 @@ on:
 jobs:
   coverage:
     name: coverage build
-    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-build-coverage.yml@master
+    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-build-coverage.yml@optional_repos
```
Comment (christophfroehlich) [inline .github/workflows/ci-coverage-build.yml:14]:
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-build-coverage.yml@master
```

Comment (christophfroehlich) [reply .github/workflows/ci-coverage-build.yml:14]: once https://github.com/ros-controls/ros2_control_ci/pull/64 is merged

Comment (saikishor) [reply .github/workflows/ci-coverage-build.yml:14]: Awesome :+1:

**Review by saikishor — APPROVED** [review]
Looking good to me. Thanks for the changes :)

File: .github/workflows/ci-coverage-build.yml [hunk 4]
```diff
@@ -11,8 +11,7 @@ on:
 jobs:
   coverage:
     name: coverage build
-    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-build-coverage.yml@master
+    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-build-coverage.yml@coverage_build
```
Comment (christophfroehlich) [inline .github/workflows/ci-coverage-build.yml:14]:
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-build-coverage.yml@master
```

File: .github/workflows/binary-build.yml [hunk 5]
```diff
@@ -20,10 +20,9 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ROS_DISTRO: [rolling, iron, humble]
+        ROS_DISTRO: [rolling, jazzy, iron, humble]
```
Comment (christophfroehlich) [inline .github/workflows/binary-build.yml:23]:
```suggestion
        ROS_DISTRO: [humble, iron, jazzy, rolling]
```

File: .github/workflows/debian-build.yml [hunk 6]
```diff
@@ -15,8 +15,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ROS_DISTRO: [rolling, iron, humble]
+        ROS_DISTRO: [rolling, iron, jazzy, humble]
```
Comment (christophfroehlich) [inline .github/workflows/debian-build.yml:18]:
```suggestion
        ROS_DISTRO: [humble, iron, jazzy, rolling]
```

File: .github/workflows/rhel-semi-binary-build.yml [hunk 7]
```diff
@@ -15,8 +15,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ROS_DISTRO: [rolling, iron, humble]
+        ROS_DISTRO: [rolling, iron, jazzy, humble]
```
Comment (christophfroehlich) [inline .github/workflows/rhel-semi-binary-build.yml:18]:
```suggestion
        ROS_DISTRO: [humble, iron, jazzy, rolling]
```

**Pull request merged**